src/indexer/demuxer.rs

```diff
@@ -0,0 +1,343 @@
+use common::BitSet;
+use itertools::Itertools;
+
+use crate::fastfield::AliveBitSet;
+use crate::{
+    merge_filtered_segments, Directory, Index, IndexSettings, Segment, SegmentOrdinal, TantivyError,
+};
+
+/// DemuxMapping can be used to reorganize data from multiple segments.
+/// e.g. if you have two tenant ids TENANT_A and TENANT_B and two segments with
```

Comment:
You start talking about tenant as if it was a concept in tantivy...
I think it requires a minimum information.


```
DemuxMapping is useful in a mutlitenant settings, in which each document might actually belong to a different tenant.
It allows to reorganize documents as follows:

The example that you give after that is good.
```